lib/sycamore/sycamore/connectors/file/file_scan.py

```diff
@@ -164,51 +192,27 @@ def execute(self, **kwargs) -> "Dataset":
 
         return files.map(self._to_document, **self.resource_args)
 
-    def local_source(self, **kwargs) -> list[Document]:
-        if isinstance(self._paths, str):
-            paths = [self._paths]
-        else:
-            paths = self._paths
-
-        documents = []
-
-        def process_file(info):
-            if not info.is_file:
-                return
-            if self._filter_paths_by_extension and not info.path.endswith(self.format()):
-                return
-
-            with self._filesystem.open_input_file(info.path) as file:
-                binary_data = file.read()
-
-            document = Document()
-            document.doc_id = str(uuid.uuid1())
-            document.type = self._binary_format
-            document.binary_representation = binary_data
-            document.properties["path"] = info.path
-            if "filetype" not in document.properties and self._binary_format is not None:
-                document.properties["filetype"] = self._file_mime_type()
-            if self._is_s3_scheme():
-                document.properties["path"] = "s3://" + info.path
-            if self._metadata_provider:
-                document.properties.update(self._metadata_provider.get_metadata(info.path))
-
-            documents.append(document)
-
-        for orig_path in paths:
-            from sycamore.utils.pyarrow import cross_check_infer_fs
+    def process_file(self, info) -> list[Document]:
+        if not info.is_file:
+            return []
+        if self._filter_paths_by_extension and not info.path.endswith(self.format()):
+            return []
 
-            (filesystem, path) = cross_check_infer_fs(self._filesystem, orig_path)
-            if self._filesystem is None:
-                self._filesystem = filesystem
+        with self._filesystem.open_input_file(info.path) as file:
+            binary_data = file.read()
 
-            path_info = filesystem.get_file_info(path)
-            if path_info.is_file:
-                process_file(path_info)
-            else:
-                for info in filesystem.get_file_info(FileSelector(path, recursive=True)):
-                    process_file(info)
-        return documents
+        document = Document()
+        document.doc_id = str(uuid.uuid1())
+        document.type = self._binary_format
+        document.binary_representation = binary_data
+        document.properties["path"] = info.path
+        if "filetype" not in document.properties and self._binary_format is not None:
+            document.properties["filetype"] = self._file_mime_type()
+        if self._is_s3_scheme():
+            document.properties["path"] = "s3://" + info.path
```

Comment:
Is this s3 stuff needed needed in the other implementations of process_file?

Reply:
I keep it the same, json_scan is handled inside the _to_document; for json_document_scan, I assume it's whatever inside the json file.